File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int currentMinutesInOven)
    {
        return ExpectedMinutesInOven() - currentMinutesInOven;
    }

    // TODO: define the 'PreparationTimeInMinutes()' method
    public int PreparationTimeInMinutes(int layers)
    {
        return layers * 2;
    }

    // TODO: define the 'ElapsedTimeInMinutes()' method
    public int ElapsedTimeInMinutes(int layers, int currentMinutesInOven)
    {
        return currentMinutesInOven + PreparationTimeInMinutes(layers);
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven curTime = expectedMinutesInOven - curTime

let preparationTimeInMinutes layers = layers * 2

let elapsedTimeInMinutes layers curTime =
    preparationTimeInMinutes layers + curTime
